### ISSUE TITLE
How to change soil moisture over long term model

### ISSUE THREAD

**Issue by env99** [APSIMInitiative/ApsimX#6067]
When running a long term model (ex// 20 year simulation), is there a way to specify the initial soil water conditions for each year if the soil moisture is different at the start of each growing season?

For example, let's say that the initial soil moisture from 0-10cm depth is:
15% in 1990
30% in 1991
23% in 1992
and so on...

Can I set these values or is there only the option to reset the soil water each year?

**Comment by hol430** [comment]
I don't think there's a way to do this built into apsim, but it should be possible via a manager script. Something like this should do what you want:

```csharp
using System;
using Models.Core;
using Models.Interfaces;
using Models.Soils;
using APSIM.Shared.Utilities;

namespace Models
{
    [Serializable]
    public class Script : Model
    {
        [Link] private ISummary summary;
        [Link] private IPhysical physical;
        [Link] private ISoilWater soilWater;

        public void ResetWater(int layer, double percentFull)
        {
        	if (percentFull < 0 || percentFull > 100)
        		throw new ArgumentException("percentFull must be in range [0, 100]");
        	if (layer >= physical.Thickness.Length || layer < 0)
        		throw new ArgumentException("layer must be in range [0, " + physical.Thickness.Length + "]");
        	soilWater.SW[layer] = MathUtilities.Divide(percentFull, 100, 0) * physical.PAWC[layer] + physical.LL15[layer];
        	soilWater.SWmm[layer] = MathUtilities.Divide(percentFull, 100, 0) * physical.PAWCmm[layer] + physical.LL15mm[layer];
        }
    }
}
```

If you put this into a manager script and then either call it from the manager script on in a daily timestep function, or from an operations node (e.g. `1900-01-05 [ResetWater].Script.ResetWater(0, 23)`). You may need to modify this code slightly if you're resetting to a proportion of something else (e.g. the plant's pawc).

**Comment by env99** [comment]
I tried using this code described above and calling it from the Operations Node, however I keep getting thrown an error that the 'layer must be in range[0,10]'. 

I then tried to substitute the 0's in the layer portion of the code with 1's since I don't think I have a 0 layer, but this threw me the same error 'layer must be in range [1,10]'.

I've attached a picture of the error message, as well as the method I used for calling the function from the Operations Node. 

Any help or suggestions would be appreciated. 
<img width="352" alt="Reset Water Operation" src="https://user-images.githubusercontent.com/74792163/107682585-c0f56c80-6c5d-11eb-890e-d6aff04e6279.PNG">
<img width="578" alt="Reset Water Error" src="https://user-images.githubusercontent.com/74792163/107682591-c357c680-6c5d-11eb-9c03-1e84f793ecee.PNG">
